app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java

```diff
@@ -325,15 +325,16 @@ protected void draw() {
 
         frame++;
         long currentTime = System.currentTimeMillis() - startTime;
-        float secondsElapsed = (currentTime) / 1000f; // will be input into shader as iTime
-        float deltaTime = (currentTime - prevTime) / 1000f;
-        float fps = 1 / deltaTime;
+        double secondsElapsed = (currentTime) / 1000.0; // will be input into shader as iTime
+        double deltaTime = (currentTime - prevTime) / 1000.0;
+        double fps = 1 / deltaTime;
         prevTime = currentTime;
-        if (secondsElapsed > 60 * 60 * 24) {
+        if (secondsElapsed > 60 * 60 * 24 || Double.isNaN(fps) || Double.isNan(deltaTime)) {
```

Comment:
As per the previous comment, this should now be Float.isNaN. Also the second call needs the last n to be capitalized.

Reply:
KKKKKKKKK